lib/src/facts/posix/xen_resolver.cc

```diff
@@ -0,0 +1,45 @@
+#include <internal/facts/posix/xen_resolver.hpp>
+#include <facter/execution/execution.hpp>
+#include <leatherman/logging/logging.hpp>
+#include <boost/filesystem.hpp>
+
+using namespace std;
+using namespace facter::facts;
+using namespace facter::execution;
+using namespace boost::filesystem;
+namespace bs = boost::system;
+
+namespace facter { namespace facts { namespace posix {
+
+    string xen_resolver::xen_command()
+    {
+        constexpr char const* xen_toolstack = "/usr/lib/xen-common/bin/xen-toolstack";
+
+        bs::error_code ec;
+        if (exists(xen_toolstack, ec) && !ec) {
+            bool success;
+            string output, error;
+            tie(success, output, error) = execute(xen_toolstack);
+            if (success) {
+                return output;
+            } else {
+                LOG_DEBUG("failure executing %1%: %2%", xen_toolstack, error);
```

Comment:
I followed the original logic that said no. I'm open to argument, but not sure what the case is for having xen-toolstack installed.